### ISSUE TITLE
Support new credential types?

### ISSUE THREAD

**Issue by bhatman** [antonvs2/gcal-cli#14]
Hi there! I think Google has changed the types of Oauth2 credentials that can be generated in the dev console -- they no longer have the "Other" type. Any current guidance on how to generate the right credentials? Thanks!

**Comment by toniov** [comment]
Hi! Thanks for raising the issue, also could check that the JSON file you get now is in a different format.

I'll release a new version fixing it.

**Comment by Japh** [comment]
@bhatman Which credential type did you select in the end? <3

https://xkcd.com/979/

**Comment by ferschubert-hm** [comment]
@Japh had the same issue today. Could fix it by using the Desktop application type.

**Comment by jerico** [comment]
There's a new issue I encountered. Even if application type is desktop, I'm blocked by:

![Screen Shot 2022-03-04 at 5 40 27 PM](https://user-images.githubusercontent.com/1062325/156743383-75f8c0df-518d-4552-a6d9-da0c3c8037d6.png)

This can be fixed by the following steps
- Choose Desktop Application when creating a token as @ferschubert-hm mentioned
- Download the token as json
- Copy to `~/client_secret.json`
- Go back to the OAuth consent screen
- Set **Publishing Status** to **Testing**
- Add your email in **Test users**
- `gcal generateUrl`
- Visit URL generated - you should be able to authenticate and get the token
